File: class08/python/AP0802-ConversaoTipos.py
# Conversao de tipos (casting)
texto = "42"
numero = int(texto)          # str -> int
print(numero + 8)            # 50

real = float("3.14")         # str -> float
print(real * 2)              # 6.28

inteiro = int(9.99)          # float -> int (trunca)
print(inteiro)               # 9

como_texto = str(2026)       # int -> str
print("Ano: " + como_texto)

File: class08/cpp/AP0802-ConversaoTipos.cpp
#include <iostream>
#include <string>
int main() {
    std::string texto = "42";
    int numero = std::stoi(texto);          // string -> int
    std::cout << numero + 8 << std::endl;   // 50

    double real = std::stod("3.14");        // string -> double
    std::cout << real * 2 << std::endl;     // 6.28

    int inteiro = (int) 9.99;               // double -> int (trunca)
    std::cout << inteiro << std::endl;      // 9

    std::string comoTexto = std::to_string(2026);  // int -> string
    std::cout << "Ano: " << comoTexto << std::endl;
    return 0;
}
